runbooks/sectigo-root-ca-rotation-2025/linux.sh

```diff
@@ -0,0 +1,181 @@
+#!/bin/bash
+# Datadog Agent v5 runbook — checks only logs since this script's restart.
+# Hardened: uses sudo where needed, avoids assoc arrays, robust log slicing.
+
+TARGET_DIR="/opt/datadog-agent/agent"
+TARGET_FILE="${TARGET_DIR}/datadog-cert.pem"
+URL="https://raw.githubusercontent.com/DataDog/dd-agent/master/datadog-cert.pem"
+CONF_FILE="/etc/dd-agent/datadog.conf"
+LOG_FILES="/var/log/datadog/forwarder.log /var/log/datadog/collector.log"
+
+TEMP_INSTALLED=0
+DOWNLOADER=""
+OFFSETS_FILE="$(mktemp -t dd_offsets_XXXXXX)"
+PRE_TS_UNIX=""
+PRE_TS_READABLE=""
+REMOVE_CMD=""
+
+error_exit() {
+  echo "$1" >&2
+  echo "Please contact support for further help." >&2
+  exit 1
+}
+
+install_curl() {
+  echo "Neither curl nor wget found. Attempting to install curl temporarily..."
+  if command -v apt-get &>/dev/null; then
+    INSTALL_CMD="sudo apt-get update && sudo apt-get install -y curl"
+    REMOVE_CMD="sudo apt-get remove -y curl"
+  elif command -v yum &>/dev/null; then
+    INSTALL_CMD="sudo yum install -y curl"
+    REMOVE_CMD="sudo yum remove -y curl"
+  elif command -v dnf &>/dev/null; then
+    INSTALL_CMD="sudo dnf install -y curl"
+    REMOVE_CMD="sudo dnf remove -y curl"
+  else
+    error_exit "Error: No supported package manager found to install curl. Please install curl or wget manually."
+  fi
+  if eval "$INSTALL_CMD"; then
+    echo "Successfully installed curl."
+    TEMP_INSTALLED=1
+    DOWNLOADER="curl"
+  else
+    error_exit "Error: Failed to install curl."
+  fi
+}
+
+check_downloader() {
+  if command -v curl &>/dev/null; then
+    DOWNLOADER="curl"
+  elif command -v wget &>/dev/null; then
+    DOWNLOADER="wget"
+  else
+    install_curl
+  fi
+}
+
+ensure_target_directory() {
+  if ! sudo test -d "$TARGET_DIR"; then
+    echo "Directory $TARGET_DIR does not exist. Creating it..."
+    sudo mkdir -p "$TARGET_DIR" || error_exit "Error: Failed to create $TARGET_DIR."
+  fi
+}
+
+download_certificate() {
+  echo "Downloading the DataDog certificate using $DOWNLOADER..."
+  if [ "$DOWNLOADER" = "curl" ]; then
+    sudo curl -fsSL "$URL" -o "$TARGET_FILE" || error_exit "Error: Failed to download certificate with curl."
+  else
+    sudo wget -qO "$TARGET_FILE" "$URL" || error_exit "Error: Failed to download certificate with wget."
+  fi
+  echo "Certificate downloaded successfully to $TARGET_FILE."
+}
+
+update_datadog_config() {
+  if ! sudo test -f "$CONF_FILE"; then
+    error_exit "Error: Configuration file $CONF_FILE not found."
+  fi
+
+  echo "Updating $CONF_FILE for use_curl_http_client..."
+  if sudo grep -q '^[[:space:]]*use_curl_http_client' "$CONF_FILE"; then
+    echo "Parameter 'use_curl_http_client' found. Setting its value to true..."
+    sudo sed -i 's/^[[:space:]]*use_curl_http_client.*/use_curl_http_client: true/' "$CONF_FILE" \
```

Comment:
this substitute will truncate the `[[:space:]]*, you'll need a capture group for the regex
```suggestion
  if sudo grep -q '^[[:space:]]*use_curl_http_client' "$CONF_FILE"; then
    echo "Parameter 'use_curl_http_client' found. Setting its value to true..."
    sudo sed -i 's/^\([[:space:]]*\)use_curl_http_client.*/\1use_curl_http_client: true/' "$CONF_FILE" \
```